sdk/core/azure-core-http-jdk-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/BodyPublisherUtils.java

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.jdk.httpclient;
+
+import com.azure.core.implementation.util.BinaryDataContent;
+import com.azure.core.implementation.util.BinaryDataHelper;
+import com.azure.core.implementation.util.ByteArrayContent;
+import com.azure.core.implementation.util.FileContent;
+import com.azure.core.implementation.util.InputStreamContent;
+import com.azure.core.implementation.util.SerializableContent;
+import com.azure.core.implementation.util.StringContent;
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.ProgressReporter;
+import com.azure.core.util.logging.ClientLogger;
+import reactor.adapter.JdkFlowAdapter;
+import reactor.core.publisher.Flux;
+
+import java.io.FileNotFoundException;
+import java.io.UncheckedIOException;
+import java.net.http.HttpRequest;
+import java.nio.ByteBuffer;
+import java.util.concurrent.Flow;
+
+import static java.net.http.HttpRequest.BodyPublishers.fromPublisher;
+import static java.net.http.HttpRequest.BodyPublishers.noBody;
+import static java.net.http.HttpRequest.BodyPublishers.ofByteArray;
+import static java.net.http.HttpRequest.BodyPublishers.ofFile;
+import static java.net.http.HttpRequest.BodyPublishers.ofInputStream;
+import static java.net.http.HttpRequest.BodyPublishers.ofString;
+
+final class BodyPublisherUtils {
+    private static final ClientLogger LOGGER = new ClientLogger(BodyPublisherUtils.class);
+
+    private BodyPublisherUtils() {
+    }
+
+    /**
+     * Creates BodyPublisher depending on underlying body content type.
+     * If progress reporter is not null, configures it to track request body upload.
+     *
+     * @param request {@link com.azure.core.http.HttpRequest} instance
+     * @progressReporter optional progress reporter.
+     * @return the request BodyPublisher
+     */
+    public static HttpRequest.BodyPublisher toBodyPublisher(com.azure.core.http.HttpRequest request, ProgressReporter progressReporter) {
+        BinaryData body = request.getBodyAsBinaryData();
+        if (body != null) {
+            BinaryDataContent bodyContent = BinaryDataHelper.getContent(body);
+            if (bodyContent instanceof ByteArrayContent) {
+                return getPublisherWithReporter(ofByteArray(bodyContent.toBytes()), progressReporter);
+            } else if (bodyContent instanceof StringContent || bodyContent instanceof SerializableContent) {
+                return getPublisherWithReporter(ofString(bodyContent.toString()), progressReporter);
+            } else if (bodyContent instanceof FileContent) {
+                try {
+                    return getPublisherWithReporter(ofFile(((FileContent) bodyContent).getFile()), progressReporter);
```

Comment:
Watch out on this as `FileContent` supports a range of the file which `ofFile` doesn't respect.